cpp/arcticdb/util/name_validation.cpp

```diff
@@ -118,8 +119,14 @@ void verify_library_path_part(const std::string& library_part, char delim) {
     }
 }
 
-void verify_library_path_on_write(const StringId& library_path) {
+void verify_library_path_on_write(const Store* store, const StringId& library_path) {
```

Comment:
Don't pass around raw pointers if you can help it, just pass in a copy of the shared pointer

Reply:
I was using shared pointer. But I switched to raw after this [comment](https://github.com/man-group/ArcticDB/pull/1481#discussion_r1557486377)